indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -654,6 +676,12 @@ private void announceTask(
     log.info("Worker[%s] reportin' for duty!", worker.getHost());
 
     try {
+      ScheduledFuture previousCleanup = removedWorkerCleanups.remove(worker.getHost());
+      if (previousCleanup != null) {
+        log.info("cancelling scheduled task cleanup for Worker[%s]", worker.getHost());
+        previousCleanup.cancel(false);
```

Comment:
Or else you can have this thread and the previousCleanup both trying to modify resources at the same time.